application_sdk/transformers/query/__init__.py

```diff
@@ -0,0 +1,359 @@
+        if is_literal:
+            return f"{column['name']} AS {column['name']}"
+        return f"{column['source_query']} AS {column['name']}"
+
+    def _get_columns(
+        self,
+        sql_template: Dict[str, Any],
+        dataframe: daft.DataFrame,
+        default_attributes: Dict[str, Any],
+    ) -> Tuple[List[str], Optional[List[Dict[str, str]]]]:
+        """Get the columns and literal columns for the SQL query.
+
+        Args:
+            sql_template (Dict[str, Any]): The SQL template
+            dataframe (daft.DataFrame): The DataFrame to get columns from
+            default_attributes (Dict[str, Any]): The default attributes to add to the SQL query
+
+        Returns:
+            A list of column expressions for the SQL query
+        """
+        columns: List[str] = []
+        literal_columns: List[Dict[str, str]] = []
+        column_names = dataframe.column_names + list(default_attributes.keys())
+        for column in sql_template["columns"]:
+            if column.get("source_columns") and (
+                all(col in column_names for col in column["source_columns"])
+            ):
+                columns.append(self._process_column(column))
+            elif column["source_query"] in column_names:
+                columns.append(self._process_column(column))
+            elif isinstance(column["source_query"], bool) or (
+                isinstance(column["source_query"], str)
+                and column["source_query"].startswith("'")
+                and column["source_query"].endswith("'")
+                and len(column["source_query"]) > 1
+            ):
+                # This is a string literal and should be added as is
+                literal_columns.append(column)
+                columns.append(self._process_column(column, is_literal=True))
+
+        return columns, literal_columns or None
+
+    def generate_sql_query(
+        self,
+        yaml_path: str,
+        dataframe: daft.DataFrame,
+        default_attributes: Dict[str, Any],
+    ) -> Tuple[str, Optional[List[Dict[str, str]]]]:
+        """
+        Generate a SQL query from a YAML template and a DataFrame.
+
+        Args:
+            yaml_path (str): The path to the YAML template
+            dataframe (daft.DataFrame): The DataFrame to reference for column names
+            default_attributes (Dict[str, Any]): The default attributes to add to the SQL query
+
+        Returns:
+            str: The generated SQL query
+        """
+        try:
+            with open(yaml_path, "r") as f:
+                sql_template = yaml.safe_load(f)
+            columns, literal_columns = self._get_columns(
+                sql_template, dataframe, default_attributes
+            )
+
+            sql_query = textwrap.dedent(f"""
+            SELECT
+                {','.join(columns)}
+            FROM dataframe
+            """)
+            return sql_query, literal_columns or None
```

Comment:
let's log the SQL Query

Reply:
Logging it [here](https://github.com/atlanhq/application-sdk/pull/423/files#diff-717fdae518f2c3d037302e58853b5601a8b1eec14fa3e0e0c067b7cc74a0fe1dR397)